### PR TITLE
Update java & lib versions, use confluentinc images in docker-compose, add devcontainer/codespace configuration for solution branch

### PR DESCRIPTION
**Pull request by RemiEven**

### CODE REVIEW

**Comment by loicmathieu** [comment]
@RemiEven, thanks for the PR.
As there is a lot of files to review, can you explain how you migrate the Quarkus version? Do you use the quarkus tooling? In this case I'll know that it's OK and didn't review everything ;)

**Comment by RemiEven** [comment]
Hi @loicmathieu !

Actually I generated new up-to-date quarkus projects with maven, then modified them to add the "business" part of the code, and tried to limit the conflicts with the existing files

A bit convoluted but with my very limited knowledge of quarkus it seemed easier that way!

File: order-api/pom.xml [hunk 1]
```diff
@@ -2,21 +2,22 @@
 <project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>fr.loicmathieu.demo.quarkus.messaging</groupId>
+  <groupId>org.acme</groupId>
```
Comment (loicmathieu) [inline order-api/pom.xml]: Please keep the same groupId as before

File: order-enhancer/pom.xml [hunk 2]
```diff
@@ -2,21 +2,22 @@
 <project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>fr.loicmathieu.demo.quarkus.messaging</groupId>
+  <groupId>org.acme</groupId>
```
Comment (loicmathieu) [inline order-enhancer/pom.xml]: Please keep the same groupId as before

File: search-api/pom.xml [hunk 3]
```diff
@@ -2,21 +2,22 @@
 <project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
   <modelVersion>4.0.0</modelVersion>
-  <groupId>fr.loicmathieu.demo.quarkus.messaging</groupId>
+  <groupId>org.acme</groupId>
```
Comment (loicmathieu) [inline search-api/pom.xml]: Please keep the same groupId as before

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Review by loicmathieu — APPROVED** [review]
LGMT, thanks

**Pull request merged**